### ISSUE TITLE
Assembly stuck using metaspades

### ISSUE THREAD

**Issue by JotaKas** [bxlab/metaWRAP#361]
I am using the metaspades inside metawrap to assembly some samples. However, some of my samples get stuck during read correction process and never finish assembling.

Apparently this is a known issue and there have been some updates by the spades developers to solve it. 

My question is if I can simply manually update the metaspades scripts used by metawrap or if the metawrap developers plan to make this update in the near future.

Thanks in advance

**Comment by ursky** [comment]
Feel free to manually update the environment; i will eventually update the version

**Comment by Holly202109** [comment]
I also met this problem. 
It is said "<jemalloc>: Error in malloc(): out of memory. Requested: 33790860608, active: 181231681536

== Error ==  system call for: "['/home/user/miniconda2/envs/metawrap/share/spades-3.13.0-0/bin/spades-hammer', '/data/FL/new_reads/ASSEMBLYBLG/metaspades/corrected/configs/config.info']" finished abnormally, err code: -6

In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru
or report an issue on our GitHub repository github.com/ablab/spades
Please provide us with params.txt and spades.log files from the output directory.

************************************************************************************************************************
*****                            Something went wrong with metaSPAdes assembly. Exiting.                           *****
************************************************************************************************************************
"
May I know how to solve this problem?

Thank you.

**Comment by ursky** [comment]
You just ran out of RAM - SPAdes memory usage scales nutoriously bad on larger assemblies. I would recommend using the megahit option which is much more efficient in that regard.

**Comment by Holly202109** [comment]
Thanks for the reply. The strange point is, I used SPAdes for treating 5 samples sequenctially. One of the largest (18G bp for forward reads, and 18G bp for reverse reads, and in the middle of the treating list ) was successfully assembled with metaspades, but the other four smaller ones failed, and the error message was given : out of memory, as I showed in the last message. When I tried it again, it said the same error. I can not figure it out what happened.

**Comment by ursky** [comment]
For what its worth, the memory usage does not technically depend on the amount of sequencing data, but on the number of unique kmers, so its possible your smaller sample actually needs more memory to assemble.

**Comment by liangjinsong** [comment]
> 
> 
> You just ran out of RAM - SPAdes memory usage scales nutoriously bad on larger assemblies. I would recommend using the megahit option which is much more efficient in that regard.

I disagree. From the log information " Requested: 33790860608, active: 181231681536", we know the active memory is bigger than Requested memory because 33790860608<181231681536.

I met the similar issue when assembling with metaSpades, and the Error information is listed below.

<jemalloc>: Error in malloc(): out of memory. Requested: 288352, active: 1042250989568

**Comment by Holly202109** [comment]
Thank you for the response. May I know how to update the SPAdes version in metawrap environment ?
Thank you.

**Comment by fanlu-f** [comment]
Thank you for the response. I also met this problem. Then, how can I update SPAdes in the metawrap environment ? If I can do this, it might solve the problem about memory .
Looking forward to your reply !

**Comment by liangjinsong** [comment]
Hi, you can update spades in the metawrap environment by the following command.


conda update spades


On 5/3/2021 ***@***.***> wrote：

Thank you for the response. I also met this problem. Then, how can I update SPAdes in the metawrap environment ? If I can do this, it might solve the problem about memory .
Looking forward to your reply !

**Comment by liangjinsong** [comment]
Hi, you can update spades in the metawrap environment by the following command.


conda update spades


On 5/3/2021 ***@***.***> wrote：

Thank you for the response. May I know how to update the SPAdes version in metawrap environment ?
Thank you.
